src/ch00-00-introduction.md

```diff
@@ -0,0 +1,382 @@
+broadly accessible to those from a wide variety of programming backgrounds. We
+don’t spend a lot of time talking about what programming _is_ or how to think
+about it. If you’re entirely new to programming, you would be better served by
+reading a book that specifically provides an introduction to programming.
+ -->
+
+## Para Quem É Este Livro
+
+Este livro assume que você tenha escrito código em outra linguagem mas não
+necessariamente assume sobre qual. Nós tentamos fazer do material amplamente
+acessível àqueles de uma grande variedade de históricos em programação. Nós
+não gastamos muito tempo falando sobre o que a programação _é_ ou como pensar
+sobre ela. Se você é inteiramente novo à programação, você seria melhor servido
+lendo um livro que especificamente provê uma introdução à programação.
+
+<!-- ## How to Use This Book
+
+In general, this book assumes that you’re reading it in sequence from front to
+back. Later chapters build on concepts in earlier chapters, and earlier
+chapters might not delve into details on a topic; we typically revisit the
+topic in a later chapter.
+
+You’ll find two kinds of chapters in this book: concept chapters and project
+chapters. In concept chapters, you’ll learn about an aspect of Rust. In project
+chapters, we’ll build small programs together, applying what you’ve learned so
+far. Chapters 2, 12, and 20 are project chapters; the rest are concept chapters.
+
+Chapter 1 explains how to install Rust, how to write a Hello, world! program,
+and how to use Cargo, Rust’s package manager and build tool. Chapter 2 is a
+hands-on introduction to the Rust language. Here we cover concepts at a high
+level, and later chapters will provide additional detail. If you want to get
+your hands dirty right away, Chapter 2 is the place for that. At first, you
+might even want to skip Chapter 3, which covers Rust features similar to those
+of other programming languages, and head straight to Chapter 4 to learn about
+Rust’s ownership system. However, if you’re a particularly meticulous learner
+who prefers to learn every detail before moving on to the next, you might want
+to skip Chapter 2 and go straight to Chapter 3, returning to Chapter 2 when
+you’d like to work on a project applying the details you’ve learned. -->
+
+## Como Utilizar Este Livro
+
+Em geral, este livro assume que você o está lendo em sequência do começo ao
+fim. Capítulos posteriores constroem sobre conceitos de capítulos anteriores,
+e capítulos anteriores podem não entrar em detalhes num tópico. Nós
+tipicamente revisitamos o tópico num capítulo posterior.
+
+Você irá encontrar dois tipos de capítulos neste livro: capítulos sobre
+conceitos e sobre projetos. Em capítulos sobre conceitos, você irá aprender
+sobre algum aspecto de Rust. Em capítulos de projetos, nós iremos construir
+pequenos programas juntos, aplicando o que você aprendeu até aqui. Os capítulos
+2, 12, e 20 são capítulos de projeto. O resto são sobre conceitos.
```

Comment:
Aqui é com ponto final mesmo, ou teria o ponto e vírgula como a versão original?

Reply:
Então, a pontuação em geral do original em inglês é bastante errática. Os autores não seguiram um padrão claro e toda hora usam uma coisa diferente. Sem contar que sinto que eles forçam um pouquinho a mão nas vírgulas.

Dito isso não acredito que a gente tenha que ter um projeto reformista e tentar domar a sintaxe do original. Me sinto como os tradutores do tolstói que ficavam corrigindo as idiossincrasias dele pra deixar o texto mais comportado. Esse é um livro técnico e não um livro de literatura, mas ainda assim. A pontuação diz muito dobre os autores.